scribble-lib/scribble/manual-racket.js

```diff
@@ -15,14 +15,175 @@ AddOnLoad(function() {
+}
+
+var prefixes = { "github.com": "tree",
+                 "gitlab.com": "-/blob" };
+
+
+function AddSourceUrl(source, mod_path, collection, info) {
+    // multi is encoded as an array, empty as false
+    single_collection = (typeof collection === "string");
+
+    var parsed = source && mod_path && ParseSource(source, mod_path, single_collection);
+
+    if (!parsed) return;
+
+    prefix = prefixes.hasOwnProperty(parsed.host) && prefixes[parsed.host];
+    if (!prefix) return;
+
+    var correct_url = "https://" + [parsed.host, parsed.user, parsed.repo, prefix, parsed.branch, parsed.file_path].join("/");
+
+    if (info) AddSourceElement(correct_url, info);
+}
+
+function addSpan(dest, str, cn) {
+    var s = document.createElement("span");
+    s.className = cn;
+    s.style.whiteSpace = "nowrap";
+    s.appendChild(document.createTextNode(str));
+    dest.appendChild(s);
+}
+
+
+// test cases
+if (false) {
+    console.log(ParseSource("git://gitlab.com/benn/foo?path=xxx",
+                            '(lib "asn1/scribblings/asn1.scrbl")',
+                            false))
+    console.log(ParseSource("github://github.com/carl-eastlund/mischief/master",
+                            '(lib "asn1/scribblings/asn1.scrbl")',
+                            false))
+    console.log(ParseSource("github://github.com/carl-eastlund/mischief/stable/dir",
+                            '(lib "asn1/scribblings/asn1.scrbl")',
+                            false))
+
+    console.log(ParseSource("git://github.com/racket/racket/?path=pkgs/racket-doc",
+                            '(lib "asn1/scribblings/asn1.scrbl")',
+                            false));
+
+    console.log(ParseSource("git://github.com/rmculpepper/asn1.git?path=asn1-doc",
+                            '(lib "asn1/scribblings/asn1.scrbl")',
+                            true));
+    console.log(ParseSource("git://github.com/rmculpepper/asn1",
+                            '(lib "asn1/scribblings/asn1.scrbl")',
+                            true));
+    console.log(ParseSource("git://github.com/rmculpepper/asn1",
+                            '(lib "asn1/scribblings/asn1.scrbl")',
+                            false));
+}
+
 function AddPartTitleOnClick(elem) {
     var mod_path = elem.getAttribute("x-source-module");
     var tag = elem.getAttribute("x-part-tag");
+    var source_pkg = elem.getAttribute("x-source-pkg");
+
+    // create here to share
+    var info = document.createElement("div");
+
+
+    // tag is not needed, but this way we can add the element in only one place
+    // avoid failing on browser that don't have `fetch`
+    if (mod_path && source_pkg && tag && window.fetch) {
+
+        var cached = cache[mod_path]
+        if (cached) {
+            AddSourceElement(cached[0], mod_path, cached[1], info);
+        }
+        else {
+            fetch("https://pkgs.racket-lang.org/pkg/" + source_pkg + ".json")
+                .then(response => response.json())
```

Comment:
It looks like you want to commit to support old browsers after all. In that case, arrow function here will cause a parsing error in old browser.